src/App/useLocalStorage.ts

```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+
+const useLocalStorage = (
+  key: string,
+  initialValue: number
+): [number, (value: number | ((val: number) => number)) => void] => {
```

Comment:
tutaj również jest inferencja - TS sam ogarnie to, jaki typ jest zwracany z funkcji - nie ma potrzeby go dodawać